lambda/shared/src/main/scala/feral/lambda/IOLambda.scala

```diff
@@ -43,3 +43,26 @@ abstract class IOLambda[Event, Result](
   def handler: Resource[IO, LambdaEnv[IO, Event] => IO[Option[Result]]]
 
 }
+
+object IOLambda {
+
+  abstract class Simple[Event, Result](
+      implicit decoder: Decoder[Event],
+      encoder: Encoder[Result])
+      extends IOLambda[Event, Result] {
+
+    type Init
+    def init: Resource[IO, Init] = Resource.pure(null.asInstanceOf[Init])
```

Comment:
`Init` instead of `Setup`, because `Setup` is already used higher up in the inheritance hierarchy.

Reply:
`Env`? `Environment`?

Reply:
Hmm. That seems more confusing. "Environment" or "Context" is usually something out of your control I think, whereas `Init` is any userland initialization.

Reply:
reasonable, Init it is then